### PR TITLE
Update the GCE scripts

### PR DESCRIPTION
**Pull request by AlCutter**
- Switch to using `HTTP Load Balancing` rather than `Network Load Balancing` since this allows the termination of HTTPS requests
- Factor out common mirror and log script.

### CODE REVIEW

File: cloud/google/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+   ```bash
+   export PROJECT="my-project"
+   make -j24
+   sudo docker build -f Dockerfile-ct-mirror -t gcr.io/${PROJECT}/super_mirror:test .
+   sudo docker build -t gcr.io/${PROJECT}/etcd:test cloud/etcd
+   gcloud docker push gcr.io/${PROJECT}/super_mirror:test
+   gcloud docker push gcr.io/${PROJECT}/etcd:test
+   ```
+
+   If using prometheus, also run:
```
Comment (phad) [inline cloud/google/README.md]: s/prom/Prom/ right?

File: cloud/google/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+1. Start mirror:
+
+   ```bash
+   ./cloud/google/create_new_cluster.sh path/to/your/config
+   ```
+
+1. Configure monitoring dashboard and alerts
+   * if using GCM click on `Monitoring > Dashboards & alerts` in your (cloud.google.com)[https://cloud.google.com]
+     console.
+   * if using prometheus configure on your prometheus instance.
```
Comment (phad) [inline cloud/google/README.md]: ditto

File: cloud/google/README.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+     You may find the following command useful while developing your prometheus
```
Comment (phad) [inline cloud/google/README.md]: etc

File: cloud/google/configure_service.sh [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+gcloud compute backend-services create "${INSTANCE_TYPE}-lb-backend" \
+    --http-health-check "get-sth-check" \
+    --timeout "30"
+
+for zone in ${ZONE_LIST}; do
+  gcloud compute backend-services add-backend "${INSTANCE_TYPE}-lb-backend" \
+    --group "${INSTANCE_TYPE}-group-${zone}" \
+    --zone ${zone} \
+    --balancing-mode "UTILIZATION" \
+    --capacity-scaler "1" \
```
Comment (phad) [inline cloud/google/configure_service.sh]: Do these tuning params deserve some explanation?  1 & 0.8

File: cloud/google/create_new_cluster.sh [hunk 5]
```diff
@@ -78,12 +88,31 @@ WaitForEtcd
...
+    ;;
+  *)
+    echo "Unknown INSTANCE_TYPE: ${INSTANCE_TYPE}"
+    exit 1
+esac
 
 echo "============================================================="
-echo "Creating superduper instances..."
-${DIR}/start_log.sh ${CONFIG_FILE}
+echo "Creating superduper ${INSTANCE_TYPE} instances..."
```
Comment (phad) [inline cloud/google/create_new_cluster.sh]: are we still referring to the current master code as superduper?

File: cloud/google/start_etcd.sh [hunk 6]
```diff
@@ -26,7 +26,12 @@ done
 wait
 
 MANIFEST=/tmp/etcd_container.yaml
-DISCOVERY=$(curl -s https://discovery.etcd.io/new?size=3)
+echo -n "Getting Discovery URL"
+while [ "${DISCOVERY}" == "" ]; do
+  DISCOVERY=$(curl -s https://discovery.etcd.io/new?size=${ETCD_NUM_REPLICAS})
```
Comment (phad) [inline cloud/google/start_etcd.sh]: Not relevant to this change, but: does etcd require some external etcd service for discovery to work?

File: cloud/google/README.md (hunk 1 above)
Comment (AlCutter) [inline cloud/google/README.md]: Yar

File: cloud/google/README.md (hunk 2 above)
Comment (AlCutter) [inline cloud/google/README.md]: ditto

File: cloud/google/README.md (hunk 3 above)
Comment (AlCutter) [inline cloud/google/README.md]: and so on

File: cloud/google/configure_service.sh (hunk 4 above)
Comment (AlCutter) [inline cloud/google/configure_service.sh]: Those are just the defaults that it gives you behind-the-scenes when you configure this via the cloud console.  There's doco for this and all other commands behind `--help` flags and on-line.

File: cloud/google/create_new_cluster.sh (hunk 5 above)
Comment (AlCutter) [inline cloud/google/create_new_cluster.sh]: It holds a special place in my heart.

File: cloud/google/start_etcd.sh (hunk 6 above)
Comment (AlCutter) [inline cloud/google/start_etcd.sh]: In the way we're doing it, yes.
There are other methods, but this one's simplest and works.

**Pull request merged**